### PR TITLE
Do a pass marking up the translations better

### LINKED ISSUE #1983: Mark all remaining translatable strings
We should mark all remaining translatable strings. In particular:
- Strings in code should be marked with `_("String")`
- Strings in HTML should be marked with the attribute `translatable="yes"` in their surrounding tag.
- Numbered translatable strings will need to use `cockpit.ngettext("Singular", "Plural", count)`, likely wrapped in a `cockpit.format(...)` call.

This will require a bit of refactoring in places:
- Where we don't have a translatable string alone in a tag, it probably needs to be put in its own tag, and then checked that this doesn't break things.
- In cases where we use Mustache we need to add translatable="yes" attributes, and then run `cockpit.translate` on the resulting element.
- In cases where we don't yet call `cockpit.translate()` on a page, we need to do that.

### LINKED ISSUE #4913: The strings with _() is not translated
In cockpit, only tanslate the strings marked with 'translatable = "yes"'.

In the Makefile.am,:
update-pofiles:
    cd $(srcdir) && \
    echo "# Generated by make update-potfiles" > po/POTFILES.in && \
    cat po/POTFILES.in.in >> po/POTFILES.in && \
    find pkg/ -name "_.js" | xargs grep -l "gettext" >> po/POTFILES.in && \
    find lib/ -name "_.js" | xargs grep -l "gettext" >> po/POTFILES.in && \
    find src/base1/ -name "_.js" | xargs grep -l "gettext" >> po/POTFILES.in && \
    find pkg/ -name "_.html" | xargs grep -l "translatable" | sed -e "s#(._)#[type: gettext/glade] \1#g" >> po/POTFILES.in && \
    find lib/ -name "_.html" | xargs grep -l "translatable" | sed -e "s#(.*)#[type: gettext/glade] \1#g" >> po/POTFILES.in

In ./src/base1/cockpit.js:
    cockpit.translate = function translate(el) {
        var list = (el || document).querySelectorAll("[translatable=\"yes\"]");
        var e, translated, i, len;
        if (list) {
            for (i = 0, len = list.length; i < len; i++) {
                e = list[i];
                translated = cockpit.gettext(e.getAttribute("context"), e.textContent);
                e.removeAttribute("translatable");
                e.textContent = translated;
            }
        }
    };

### PR DESCRIPTION
**Pull request by stefwalter**
This is likely not perfect with some ommissions. But ideally there
will be further follow ups to this to fix specific issues as they
are noted.

Fixes #4913 
Fixes #1983

### CODE REVIEW

File: pkg/machines/hostvmslist.jsx [hunk 1]
```diff
@@ -180,6 +182,7 @@ VmOverviewTabRecord.propTypes = {
     value: PropTypes.string.isRequired
 }
 
+xxxxx
```
Comment (petervo) [inline pkg/machines/hostvmslist.jsx]: typo?

File: pkg/systemd/index.html [hunk 2]
```diff
@@ -485,11 +483,8 @@ <h4 class="modal-title"></h4>
                     </table>
                 </div>
                 <div class="modal-footer">
-                    <button class="btn btn-default" translatable="yes" data-dismiss="modal">
-                        Cancel
-                    </button>
-                    <button class="btn btn-danger">
-                    </button>
+                    <button class="btn btn-default btn-danger" translatable="yes" data-dismiss="modal">Cancel</button>
```
Comment (stefwalter) [inline pkg/systemd/index.html]: Thanks I'll remove this. But unless something is actually wrong, I would suggest doing further fixes or missed translations as additional follow ups.

File: pkg/machines/hostvmslist.jsx (hunk 1 above)
Comment (stefwalter) [inline pkg/machines/hostvmslist.jsx]: Thanks. This is actually wrong.

File: pkg/systemd/index.html (hunk 2 above)
Comment (petervo) [inline pkg/systemd/index.html]: I think this breaks tests.

Comment (stefwalter) [inline pkg/systemd/index.html]: Whoops. Will fix.

**Pull request closed without merging**